tests/test_ble_utils_service_targets.py

```
@@ -0,0 +1,836 @@
+        interface=SimpleNamespace(_is_connection_closing=False, _can_initiate_connection=True),
+    )
+    hook = scheduler._resolve_thread_coordinator_hook("create_thread", "_create_thread")
+    assert callable(hook)
+
+    scheduler.thread_coordinator = SimpleNamespace()
+    with pytest.raises(ThreadCoordinatorMissingMethodError):
+        scheduler._resolve_thread_coordinator_hook("create_thread", "_create_thread")
+
+
+def test_reconnection_schedule_thread_not_started_and_policy_missing_method(
+    monkeypatch: pytest.MonkeyPatch,
```

Ruff (0.15.6)

[warning] Unused function argument: `monkeypatch`

(ARG001)